### PR TITLE
Add Cucumber features for new RespondTo matcher methods.

### PR DESCRIPTION
**Pull request by sleepingkingstudios**
- Fix failure message bug when expecting 3 or more keywords.

### CODE REVIEW

File: features/built_in_matchers/respond_to.feature [hunk 1]
```diff
@@ -92,3 +92,185 @@ Feature: `respond_to` matcher
...
+      """
+    When I run `rspec respond_to_matcher_argument_range_checking_spec.rb`
+    Then the output should contain all of these:
+      | 8 examples, 4 failures                                          |
+      | expected my_object not to respond to :build with 1..2 arguments |
+      | expected my_object to respond to :build with 0..1 arguments     |
+      | expected my_object to respond to :build with 2..3 arguments     |
+      | expected my_object to respond to :build with 0..3 arguments     |
+
+  @skip-when-splat-args-unsupported
```
Comment (JonRowe) [inline features/built_in_matchers/respond_to.feature]: This tag mentions splat arguments, but as far as I know that's always supported, and you're checking keyword arguments, can you explain further?

File: features/built_in_matchers/respond_to.feature [hunk 2]
```diff
@@ -92,3 +92,185 @@ Feature: `respond_to` matcher
...
       | expected 7 to respond to :between? with 7 arguments     |
       | expected 7 not to respond to :between? with 2 arguments |
+
+  @skip-when-splat-args-unsupported
+  Scenario: specify arguments range
+    Given a file named "respond_to_matcher_argument_range_checking_spec.rb" with:
+      """ruby
+      class MyClass
+        def build(name, options = {})
+
```
Comment (JonRowe) [inline features/built_in_matchers/respond_to.feature]: The whitespace in the method definitions isn't necessary

File: features/built_in_matchers/respond_to.feature (hunk 1 above)
Comment (sleepingkingstudios) [inline features/built_in_matchers/respond_to.feature]: I've fixed the tag helper to use the right feature. Handling optional and splat arguments is weird in 1.8.7 compared to modern Rubies, and the `::RSpec::Support::RubyFeatures.optional_and_splat_args_supported?` evaluates to false on 1.8.7.

File: features/built_in_matchers/respond_to.feature [hunk 3]
```diff
@@ -133,7 +133,7 @@ Feature: `respond_to` matcher
   Given a file named "respond_to_matcher_unlimited_argument_checking_spec.rb" with:
       """ruby
       class MyClass
-        def greet(message = 'Hello', *people)
+        def greet(message='Hello', *people)
```
Comment (JonRowe) [inline features/built_in_matchers/respond_to.feature]: I meant the blank lines, the white space in the arguments is necessary per our style guidelines! Sorry!

**Comment by JonRowe** [comment]
Merging despite build failure as thats a weird ruby warning caused by a merge from rspec/rspec-core#2271 (/cc @myronmarston)

**Comment by JonRowe** [comment]
Thanks @sleepingkingstudios !

**Comment by urbanautomaton** [comment]
@JonRowe - FWIW in rspec-core we noticed that the 2.2 build was running against 2.2.0-p0, while the warning triggered has been removed in the later 2.2.x branches. So we [forced the travis build for rspec-core to use 2.2.5](https://github.com/rspec/rspec-core/pull/2271/commits/4add3689478b05aef1e08b5515955499e833077b), which fixed the warnings.

**Comment by JonRowe** [comment]
@urbanautomaton ya I figured when I saw @myronmarston's PRs

**Pull request merged**